kafka/client.go

```diff
@@ -436,58 +454,57 @@ func isPartitionRFChanging(status *sarama.PartitionReplicaReassignmentsStatus) b
 	return len(status.AddingReplicas) != 0 || len(status.RemovingReplicas) != 0
 }
 
-func (client *Client) ReadTopic(name string) (Topic, error) {
+func (client *Client) ReadTopic(name string, refresh_metadata bool) (Topic, error) {
 	c := client.client
 
 	topic := Topic{
 		Name: name,
 	}
 
-	err := c.RefreshMetadata()
-	if err != nil {
-		log.Printf("[ERROR] Error refreshing metadata %s", err)
-		return topic, err
+	if refresh_metadata {
```

Comment:
When reading topics, sometimes just use the cached metadata rather than refreshing it, which is VERY slow when you have a lot of topics to check and a lot of topics in your cluster.

Design problems in the underlying sarama library prevent refreshing the metadata of just one topic.